habitat-lab/habitat/tasks/rearrange/social_nav/oracle_social_nav_actions.py

```diff
@@ -0,0 +1,355 @@
+        vector_human_robot = human_pos[[0, 2]] - robot_pos[[0, 2]]
+        vector_human_robot = vector_human_robot / np.linalg.norm(
+            vector_human_robot
+        )
+        forward_robot = np.array(base_T.transform_vector(mn.Vector3(1, 0, 0)))[
+            [0, 2]
+        ]
+        forward_robot = forward_robot / np.linalg.norm(forward_robot)
+        facing = np.dot(forward_robot, vector_human_robot) > 0.5
+
+        # Use geodesic distance here
+        dis = self._sim.geodesic_distance(robot_pos, human_pos)
+
+        # np.linalg.norm((robot_pos - human_pos)[[0, 2]])
+        return dis <= 2.0 and facing
+
+    def _compute_robot_to_human_min_step(
+        self, robot_trans, human_pos, human_pos_list
+    ):
+        _vel_scale = 10.0
+
+        # Copy the robot transformation
+        base_T = mn.Matrix4(robot_trans)
+
+        vc = SimpleVelocityControlEnv()
+
+        # Compute the step taken to reach the human
+        robot_pos = np.array(base_T.translation)
+        robot_pos[1] = human_pos[1]
+        step_taken = 0
+        while (
+            not self._reach_human(robot_pos, human_pos, base_T)
+            and step_taken <= 1500
+        ):
+            path_points = self._find_path_given_start_end(robot_pos, human_pos)
+            cur_nav_targ = path_points[1]
+            obj_targ_pos = path_points[1]
+            forward = np.array([1.0, 0, 0])
+            robot_forward = np.array(base_T.transform_vector(forward))
+
+            # Compute relative target.
+            rel_targ = cur_nav_targ - robot_pos
+            rel_pos = (obj_targ_pos - robot_pos)[[0, 2]]
+
+            # Compute heading angle (2D calculation)
+            robot_forward = robot_forward[[0, 2]]
+            rel_targ = rel_targ[[0, 2]]
+            angle_to_target = get_angle(robot_forward, rel_targ)
+            dist_to_final_nav_targ = np.linalg.norm(
+                (human_pos - robot_pos)[[0, 2]]
+            )
+
+            if dist_to_final_nav_targ < self._config.dist_thresh:
+                # Look at the object
+                vel = OracleNavAction._compute_turn(
+                    rel_pos,
+                    self._config.turn_velocity * _vel_scale,
+                    robot_forward,
+                )
+            elif angle_to_target < self._config.turn_thresh:
+                # Move towards the target
+                vel = [self._config.forward_velocity * _vel_scale, 0]
+            else:
+                # Look at the target waypoint.
+                vel = OracleNavAction._compute_turn(
+                    rel_targ,
+                    self._config.turn_velocity * _vel_scale,
+                    robot_forward,
+                )
+
+            # Update the robot's info
+            base_T = vc.act(base_T, vel)
+            robot_pos = np.array(base_T.translation)
+            step_taken += 1
+
+            robot_pos[1] = human_pos[1]
+        return step_taken
+
+    def _get_target_for_coord(self, obj_pos):
```

Comment:
same function as above?

Reply:
I think one needs to record the a sequence of waypoints, and the other does not. Thanks!